File: docs/ctf/scripts/_frida/windows/02_comm/communication.ts
// 注意驼峰命名 python处是 test_comm
rpc.exports = {
  testComm() {
    var data = Process.enumerateModules()
    send(data)
    recv(function (received_object) {
      var my_data = received_object.my_data
      console.log(my_data)
    }).wait()
    console.log("bye~")
  },
}

File: docs/ctf/scripts/_frida/windows/02_comm/communication.py
# 1.启动 TestAdd.exe
# 2.python .\communication.py TestAdd.exe
# 注意驼峰命名 python处是 test_comm js是 testComm
import codecs
import frida
import sys

def on_message(message, data):
    if message['type'] == 'send':
        modules = message['payload']
        target = ""
        for mod in modules:
            if sys.argv[1] in mod['name']:
                target = str(mod)
            break
        script.post({'my_data': target})
    elif message['type'] == 'error':
        print(message['stack'])

session = frida.attach(sys.argv[1])
with codecs.open('./communication.ts', 'r', 'utf-8') as f:
    source = f.read()
script = session.create_script(source)
script.on('message', on_message)
script.load()
script.exports.test_comm()